src/org/traccar/reports/ReportUtils.java

```diff
@@ -220,7 +220,12 @@ private static boolean isMoving(ArrayList<Position> positions, int index,
                 >= tripsConfig.getMinimalNoDataDuration()) {
             return false;
         }
-        return positions.get(index).getSpeed() > speedThreshold;
+        if (positions.get(index).getAttributes().containsKey(Position.KEY_MOTION)
```

Comment:
We should keep it backward compatible for about a year and then fully migrate to the new version. Please create an issue so we don't forget about it.